guides/v2.2/frontend-dev-guide/layouts/xml-manage.md

```diff
@@ -345,56 +344,143 @@ This would make the product page look like following:
+{
+    public function __construct()
+    {
 
-This block is declared in `app/code/Magento/Catalog/view/frontend/layout/default.xml`:
+    }
+}
+```
+## Modify functionality with plugins (interceptors) {#layout_markup_modify_with_plugins}
+
+To substitute or extend the behavior of original, public methods for any class or interface, we can make use of plugins, or interceptors, which are classes that modify the behavior of public class functions by intercepting a function call and running code before, after, or around that function call in the form of listeners. 
+
+This interception approach reduces conflicts among extensions that change the behavior of the same class or method, with a Plugin class implementation changing only the behavior of a class function, rather than overriding the entire class.
+
+In order to use plugins (interceptors), we must first define them in the di.xml of the module:
 
 ```xml
-<page xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:framework:View/Layout/etc/page_configuration.xsd">
-    <body>
-...
-        <referenceContainer name="sidebar.additional">
-            <block class="Magento\Catalog\Block\Product\Compare\Sidebar" name="catalog.compare.sidebar" template="product/compare/sidebar.phtml"/>
-        </referenceContainer>
-...
-    </body>
-</page>
+<config>
+    <type name="{ObservedType}">
+      <plugin name="{pluginName}" type="{PluginClassName}" sortOrder="1" disabled="false" />
+    </type>
+</config>
 ```
 
+Before listeners are used whenever we want to change the arguments of an original method or wish to add additional behavior before an original method is called. They do not need a return value.
 
-To remove the block, add the extending `default.xml` in your theme:
-`<theme_dir>/Magento_Catalog/layout/default.xml`
+Around listeners are used when we want to change both the arguments and the returned values of an original method or add new behavior before and after an original method is called. Because of this, return values are required.
 
-In this file, reference the element having added the `remove` attribute:
+After listeners are used when we want to change the values returned by an original method or want to add some behavior after an original method is called. As such, they also require return values.
+
+Let us say we want to change the behavior of an addProduct method in the Magento\Checkout\Model\Cart module by adding plugins. We first define the di.xml of this module as the following:
 
 ```xml
-<page xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:framework:View/Layout/etc/page_configuration.xsd">
-    <body>
-...
-        <referenceBlock name="catalog.compare.sidebar" remove="true" />
-...
-    </body>
-</page>
+<config>
+    <type name="Magento\Checkout\Model\Cart">
+        <plugin name="MagentoCart" type="Magento\Test\Model\Cart" sortOrder="1" />
```

Comment:
I propose removing this `sortOrder` as well. Further, would there be an alternate namespace we could use. In Magento 2, `Test` means something specific, and it would be nice to avoid that. What about: `Company\Sample`? Please change all subsequent uses as well.